src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -180,6 +184,9 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count, Tas
 	EState *estate = queryDesc->estate;
 	CmdType operation = queryDesc->operation;
 	MemoryContext oldcontext = NULL;
+	DestReceiver *destination = queryDesc->dest;
+	MaterialState *routerState = (MaterialState *) queryDesc->planstate;
+	bool sendTuples = operation == CMD_SELECT || queryDesc->plannedstmt->hasReturning;
```

Comment:
Minor: Is it `sendTuples` or, `receiveTuples` would fit better? 
Plus, I'd prefer to have an explicit `if` to assign the value to the variable. But, feel free to overlook this comment.